azurerm/internal/services/frontdoor/helper.go

```diff
@@ -201,3 +203,90 @@ func FlattenFrontendEndpointLinkSlice(input *[]frontdoor.FrontendEndpointLink) [
 	}
 	return result
 }
+
+// ParseAzureResourceIDLowerPath converts a long-form Azure Resource Manager ID
+// into a ResourceID. We make assumptions about the structure of URLs,
+// which is obviously not good, but the best thing available given the
+// SDK. I had to normalize the key casing of Path because the Front Door API
+// via Portal does not have consistent casing within the resource, for example:
+//
+// In the backendPools block the casing of the HealthProbeSettings is (notice the lowercase 'h'):
+// portal-front-door/ -> healthProbeSettings/healthProbeSettings-1571100669337
+//
+// but in the HealthProbeSettings block the casing of the HealthProbeSettings is (notice the uppercase 'H')::
+// portal-front-door/ -> HealthProbeSettings/healthProbeSettings-1571100669337
```

Comment:
it'd be worth raising this as a bug in the API - since the ARM Specification states that the same Resource ID should be returned in the same casing for the same API call, wherever it's present.

It's probably also worth calling out specifically that whilst the ARM Specification "technically" says Resource ID's are case insensitive; in practice they're not since both downstream API's and downstream client's don't consume them in that fashion - which I think is a larger discussion we need to have with the ARM Team 🙃